nixos/modules/services/logging/packetbeat.nix

```diff
@@ -0,0 +1,269 @@
+            };
+
+            packetbeat = {
+              interfaces = mkOption {
+                type = format.type;
+                description = ''
+                  Configuration of what interfaces packetbeat should monitor and how. See
+                  <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-interfaces.html'/>
+                  for all available configuration options.
+                '';
+              };
+              flows = mkOption {
+                type = format.type;
+                description = ''
+                  Configuration of how packetbeat should handle flows. See
+                  <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-flows.html'/>
+                  for all available configuration options.
+                '';
+              };
+
+              protocols = mkOption {
+                type = types.listOf format.type;
+                description = ''
+                  Configuration of what protocols packetbeat should gather info about.
+                  See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-protocols.html'/>
+                  for the configuration options available.
+                '';
+              };
+            };
+            output = mkOption {
+              type = format.type;
+              description = ''
+                Configuration of where packetbeat should send the information its gathered.
+                See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuring-output.html'/>
+                for the configuration options available.
+              '';
+            };
+
+            setup = mkOption {
+              type = format.type;
+              description = ''
+                Configures where the kibana endpoint is for setting up dashboards and
+                whether packetbeat should setup ILM and index templates in elasticsearch.
+              '';
+            };
+
+            fields = mkOption {
+              type = format.type;
+              description = ''
+                Extra fields that packetbeat should add to the top-level of the objects it
+                sends to its output.
+              '';
+            };
+
+            processors = mkOption {
+              type = format.type;
+              description = ''
+                Configuration of what enrichment and/or filtering processors packetbeat
+                should gather information from before sending it to the output.
+                See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/filtering-and-enhancing-data.html'/>
+                for the configuration options available.
+              '';
+            };
+          };
+        };
+        default = {};
+        description = ''
+          Any other configuration options you want to add.
```

Comment:
Upstream docs for this option would be nice.

Reply:
I added it in the latest commit